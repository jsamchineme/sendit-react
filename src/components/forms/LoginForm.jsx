import React from 'react'
import { Field, reduxForm } from 'redux-form';
import { NavLink } from 'react-router-dom';
import { userLoginRules } from './rules/formRules';


const renderField = ({
  input,
  label,
  type,
  meta: { touched, error }
}) => {
  return (
    <div>
      <label>{label}</label>
      {touched &&
        ((error && <div className="error-box active">{error}</div>))}
      <div>
        <input {...input} placeholder={label} type={type} />
      </div>
    </div>
  )
}


const LoginForm = props => {
  const { handleSubmit, serverErrorMessage, pristine, reset, submitting } = props;
  return (
    <form onSubmit={handleSubmit}>
      <div className="form-header">
        Login
        {serverErrorMessage &&
        <div id="server-error-box" className="error-box active">{serverErrorMessage}</div>}
      </div>
      <div className="form-body">
        <div className="input-group transparent-box">
          <Field
            name="email"
            type="email"
            component={renderField}
            label="email"
            validate={userLoginRules.email}
          />
          <Field
            name="password"
            type="password"
            component={renderField}
            label="password"
            validate={userLoginRules.password}
          />
        </div>
        <div className="input-footer" />
        <div className="row text-center">
          <div className="column">
            <button className="btn submit">Login</button>
          </div>
          <div className="actions column text-center">
            <div className="v-gap-2" />
            <div>
              <NavLink to='/signup'>Signup</NavLink>
              <span style={{marginRight: 10, display: 'inline-block'}} />
              <NavLink to='/forgot-password'>Forgot Password</NavLink>
            </div>
          </div>
        </div>
      </div>
    </form>
  )
}

export default reduxForm({
  form: 'loginForm'
})(LoginForm);