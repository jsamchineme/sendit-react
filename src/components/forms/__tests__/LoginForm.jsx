import React from 'react';
import { shallow, mount } from 'enzyme';
import { LoginForm } from '../LoginForm';
import configureStore from '../../../redux/configureStore';

const store = configureStore();

describe('Tests for login form', () => {
  it('should render the login form', async () => {
    const props = {
      handleSubmit: jest.fn(), 
      serverErrorMessage: '',
    }
    const wrapper = shallow(<LoginForm  {...props} />);

    // await wrapper.find('input').at(0).simulate('change', {
    //   target: { value: 'asdasd' }
    // });

    // console.log(wrapper.props());

    // expect(wrapper.state());

    expect(wrapper).toMatchSnapshot();
  });
});