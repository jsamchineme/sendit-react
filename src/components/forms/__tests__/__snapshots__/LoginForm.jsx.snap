// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for login form should render the login form 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="loginForm"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
